### ISSUE TITLE
Unclear documentation of 'web3.eth.sendSignedTransaction'

### ISSUE THREAD

**Issue by barakman** [ethereum/web3.js#3236]
From [the official documentation](https://web3js.readthedocs.io/en/v1.2.4/web3-eth.html#sendsignedtransaction) of `web3.eth.sendSignedTransaction`:

First, there is this description of the `callback` parameter:
> Optional callback, returns an error object as first parameter and the result as second.

Then, there is this coding example:
```
web3.eth.sendSignedTransaction('0x' + serializedTx.toString('hex'))
.on('receipt', console.log);
```

First of all, this coding "shortcut" yields a poor example for the given function (as I have previously explain in [this similar issue](https://github.com/ethereum/web3.js/issues/2724)).
It is simply unclear how exactly the input parameters of the callback function are "embedded" here.

Second, from this coding example, it seems that the callback function takes only one parameter (which is passed directly to `console.log`).

If my second observation is correct, then not only is this a poor coding example, but also the initial description of the callback function (at the top of this post) is wrong.

Thanks

Update:

Perhaps I was wrong in thinking that the callback function is used here to begin with, in which case, there is no coding example for how to pass this function as input to `web3.eth.sendSignedTransaction`.

In any case, I still think that the `.on('receipt', console.log)` syntax doesn't achieve the purpose of a good coding example.

Thanks 2 :)

**Comment by lautarodragan** [comment]
@barakman see https://github.com/ethereum/web3.js/issues/1547#issuecomment-558618977. If I understand correctly, the callback will be entirely gone in the next major release and the PromiEvent returned by it will be improved.

Under the 1.x branch, I've found BOTH the callback and `.on('error'` are needed - in some cases I noticed the callback was being called with an error but the `.on('error'` wasn't being triggered. From the callback I'm only paying attention to the error, and using the `.on('receipt'`, `.on('confirmation'` and `.on('transactionHash'` for everything else.

Not sure if this is the best approach, but it has worked for me for now (haven't released to prod yet though).
